### PR TITLE
Feature/add new sorting

### PR DESCRIPTION
**Pull request by JoeyStewart**
What I did:

I added new sorting features that are done by state, park designation, or both. Minimal styling was added for the mean time. Since I created a new branch and it is not clear where I have made changes, I will provide the lines to where my changes mostly occurred within ParksChecklist.js. 
lines 21-25, 96-131, and 158-267. 

Did this break anything?
- [x] No
- [ ]  Yes
Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x]  Styling
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.
Checklist:
- [x]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally
- [ ]  There are comments where clarification/ organization is needed
- [x]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing
What's next?

### CODE REVIEW

**Review by tenthwalker — APPROVED** [review]
Looks good, the overlapping filters are a good choice for further improving user experience.

File: src/ParkChecklist/ParkChecklist.css [hunk 1]
```diff
@@ -106,7 +106,55 @@
...
+   }
+   
+   .dropdown-menu-state label,
+   .dropdown-menu-des label {
+    display: flex;
+    align-items: center;
+    margin-bottom: 5px;
+   }
+   
+   .dropdown-menu-state input,
```
Comment (Jesuitman) [inline src/ParkChecklist/ParkChecklist.css:145]: i love that you made all these new css rules! this looks fantastic! was it difficult to implement? I'm curious about the <input> section of these rules. There's so many things in css that i just learn from exposure to other peoples code so I'm really excited to dig into this

File: src/ParkChecklist/ParkChecklist.js [hunk 2]
```diff
@@ -113,20 +155,115 @@ const ParkChecklist = ({ apiLink }) => {
...
+                    ))}
+                  </div>
+                </div>
+              </label>
+            </div>
+            <div className="sort-by-designation">
+              <label>
+                <div
+                  className={`dropdown ${dropdownDesignationOpen ? "open" : ""}`}
+                  onClick={toggleDropdownDesignation}
```
Comment (Jesuitman) [inline src/ParkChecklist/ParkChecklist.js:187]: holy god thats a lot of code to have to write for sorting, good on you for getting it pumped out!

File: src/ParkChecklist/ParkChecklist.js [hunk 3]
```diff
@@ -113,20 +155,115 @@ const ParkChecklist = ({ apiLink }) => {
...
+                      (designation) =>
+                        park.attributes.designation === designation
+                    )
+                )
+                .map((park) => (
+                  <li key={park.id}>
+                    <input
+                      type="checkbox"
+                      id={park.id}
+                      checked={checkedItems.includes(park.id)}
```
Comment (Jesuitman) [inline src/ParkChecklist/ParkChecklist.js:230]: im really excited to try this!

**Review by Jesuitman — APPROVED** [review]

**Pull request merged**